runtime_lib/airhost/memory.cpp

```diff
@@ -210,199 +210,25 @@ uint64_t air_dev_mem_get_pa(void *buff_va) {
-    for (int n=0; n<length; n+=elem_per_stride) {
-      // This is the input, so we need to take what is in t and put it into the BRAM
-      memcpy(data_ptr, (void*)bounce_ptr, elem_per_stride*sizeof(uint32_t));
-      data_ptr += stride;
-      bounce_ptr += elem_per_stride;
-    }
-  }
-}
-
-void air_shim_memcpy_queue_impl(signal_t *s, uint32_t id, uint64_t x,
-                                uint64_t y, void *t, uint64_t offset,
-                                uint64_t length) {
-  assert(_air_host_active_herd.herd_desc && "cannot shim memcpy without active herd");
-  assert(_air_host_active_herd.q && "cannot shim memcpy using a queue without active queue");
-
-  auto shim_desc = _air_host_active_herd.herd_desc->shim_desc;
-  auto shim_col = shim_location_data(shim_desc, id-1, x, y);
-  auto shim_chan = shim_channel_data(shim_desc, id-1, x, y);
-
-  tensor_t<uint32_t,1> *tt = (tensor_t<uint32_t,1> *)t;
-  uint32_t *bounce_buffer = _air_host_bram_ptr;
-  bool isMM2S = shim_chan >= 2;
-
-  if (isMM2S) {
-    shim_chan = shim_chan - 2;
-    uint32_t *data_ptr = tt->data + offset;
-    // This is the input, so we need to take what is in t and put it into the BRAM
-    memcpy((void*)bounce_buffer, data_ptr, length*sizeof(uint32_t));
-  }
-
-  uint64_t wr_idx = queue_add_write_index(_air_host_active_herd.q, 1);
-  uint64_t packet_id = wr_idx % _air_host_active_herd.q->size;
-
-  dispatch_packet_t *pkt = (dispatch_packet_t*)(_air_host_active_herd.q->base_address_vaddr) + packet_id;
-  air_packet_nd_memcpy(pkt, /*herd_id=*/0, shim_col, /*direction=*/isMM2S, shim_chan, /*burst_len=*/4, /*memory_space=*/2,
-                       _air_host_bram_paddr, length*sizeof(uint32_t), 1, 0, 1, 0, 1, 0);
-  if (s) {
-    // TODO: don't block here
-    air_queue_dispatch_and_wait(_air_host_active_herd.q, wr_idx, pkt);
-    uint64_t signal_offset = offsetof(dispatch_packet_t, completion_signal);
-    s->handle = queue_paddr_from_index(_air_host_active_herd.q,
-                                       (packet_id) * sizeof(dispatch_packet_t) +
-                                           signal_offset);
-  } else {
-    air_queue_dispatch_and_wait(_air_host_active_herd.q, wr_idx, pkt);
-  }
-  if (!isMM2S) {
-    uint32_t *data_ptr = tt->data + offset;
-    memcpy(data_ptr, (void*)bounce_buffer, length*sizeof(uint32_t));
-  }
-}
-
-template<typename T0, int R0, typename T1, int R1>
-void air_mem_cdma_nd_memcpy_queue_impl(tensor_t<T0, R0>* t0, tensor_t<T1, R1>* t1, uint32_t space0, uint32_t space1, 
-                                 uint64_t offset_3, uint64_t offset_2, uint64_t offset_1, uint64_t offset_0,
-                                 uint64_t length_4d, uint64_t length_3d, uint64_t length_2d, uint64_t length_1d,
-                                 uint64_t stride_4d, uint64_t stride_3d, uint64_t stride_2d) {
+static template <typename T0, int R0, typename T1, int R1>
+void air_mem_cdma_nd_memcpy_queue_impl(tensor_t<T0, R0> *t0,
```

Comment:
This could be removed. None of our current platforms will really make use of this. It was mainly for L3/L2 transfers emulating MemTiles in URAM.